doc/user_mgmt/user_mgmt_hld.md

```diff
@@ -0,0 +1,595 @@
+# User Management design #
+
+##  1. Table of Content
+
+
+* 1. [Table of Content](#TableofContent)
+	* 1.1. [Revision](#Revision)
+	* 1.2. [Scope](#Scope)
+	* 1.3. [Definitions/Abbreviations](#DefinitionsAbbreviations)
+	* 1.4. [Overview](#Overview)
+	* 1.5. [Requirements](#Requirements)
+		* 1.5.1. [Define defaults](#Definedefaults)
+		* 1.5.2. [Configurations](#Configurations)
+* 2. [Design](#Design)
+	* 2.1. [Architecture Design](#ArchitectureDesign)
+	* 2.2. [High-Level Design](#High-LevelDesign)
+	* 2.2. [Usercfd Role](#UsercfdRole)
+* 3. [Configuration and management](#Configurationandmanagement)
+		* 3.1. [ConfigDB Tables](#ConfigDBTables)
+		* 3.2. [ConfigDB schemas](#ConfigDBschemas)
+		* 3.3. [CLI/YANG model](#CLIYANGmodel)
+* 4. [Build](#Build)
+		* 4.1. [Compilation](#Compilation)
+		* 4.2. [Feature default](#Featuredefault)
+* 5. [Flows](#Flows)
+		* 5.1. [Init Flow](#InitFlow)
+		* 5.2. [Upgrade Flow](#UpgradeFlow)
+		* 5.3. [User add/delete](#UserAddDelete)
+		* 5.4. [User state](#UserState)
+		* 5.5. [User full-name](#UserFullName)
+		* 5.6. [User role](#UserRole)
+		* 5.7. [User hashed-password](#UserHashedPassword)
+        * 5.8. [User password expiration](#UserPasswordExpiration)
+        * 5.9. [User passwdrd prompt](#UserPasswdrdPrompt)
+* 6. [SAI API](#SAIAPI)
+* 7. [Warmboot and Fastboot Design Impact](#WarmbootandFastbootDesignImpact)
+* 8. [Restrictions/Limitations](#RestrictionsLimitations)
+* 9. [Test Plan](#TestPlan)
+		* 9.1. [Unit Test cases](#UnitTestcases)
+
+
+
+###  1.1. Revision
+|  Rev  |  Date   |      Author      | Change Description |
+| :---: | :-----: | :--------------: | ------------------ |
+|  0.1  | 12/2022 | Mohammed Zayadna | Phase 1 Design     |
+
+###  1.2. <a name='Scope'></a>Scope
+
+This HLD document described the requirements, architecture and configuration details of User Management feature in switches Sonic OS based.
+
+###  1.3. <a name='DefinitionsAbbreviations'></a>Definitions/Abbreviations
+Role - capability of user
+
+
+###  1.4. <a name='Overview'></a>Overview
+
+This document provides high level design for the username mgmt and configuration in Sonic.<br/>
+As of today, Sonic doesn’t have special configurations and management for local users. Users can be configured only by regular Linux commands.<br/>
+The goal is to standardize and define the User management configurations and functionality.<br/>
+It will define the default roles (capabilities), default users and describe the supported configurations
+for users.<br/>
+
+Note:
+The compilition of the feature code will be controlled by a build flag and it will be disabled by default.
+
+###  1.5. <a name='Requirements'></a>Requirements
+####  1.5.1. <a name='Definedefaults'></a>Define defaults
+* Default roles: admin and monitor<br/>
```

Comment:
Propose updating the roles to `admin` and `operator` (instead of `monitor`). I *think* the `operator` term is more widely used for the least-privileged / read-only role in other RBAC implementations, but I might be biased by my own experience at Dell.

Reply:
operators usually have limited write permissions. our monitor users will have read-only permissions
Please see this RBAC document: [RBAC](https://docs.jboss.org/author/display/WFLY9/RBAC.html)
Monitor – a read-only role. Cannot modify any resource.
Operator – Monitor permissions, plus can modify runtime state, but cannot modify anything that ends up in the persistent configuration. Could, for example, restart a server.


I found a similar description of operator in Dell website:
[Dell users](https://infohub.delltechnologies.com/l/bare-metal-orchestrator-1-4-command-line-interface-reference-guide/users-and-roles-overview-4)